### PR TITLE
Optimize evaluator

### LINKED ISSUE #1597: GUI lags for larger experiments
**Describe the bug**
Seems the raw flow of events from the dispatchers is too much for the GUI to handle for the moment. All dispatcher events will produce a snapshot update event, which then will be handled by the GUI.

It's not necessary to update the GUI so often. The CLI will also be spammed, since it basically `print` on ever update event.

**To Reproduce**
1. Make 100 poly_eval.py jobs
1. Set MAX_RUNNING to 1
1. `ert gui my_config.ert`
1. Run ensemble experiment (a couple of reals is fine)

**Expected behavior**
It's slow, and if you look at `status.json`, the reals will be done long before they are done in the GUI.

### PR DESCRIPTION
**Pull request by jondequinor**
**Issue**
Resolves #1597 

**Approach**
- Make sure `PartialSnapshot.from_cloudevent` can be called multiple times and does the right thing
- Ever _n_ second, consume events from the monitor and merge all `SnapshotUpdateEvent`, and emit that to the UIs. If the end, or a new iteration, is encountered, a merged update event is always dispatched. So the batched update events are emitted either every _n_ seconds; when the iteration ends; or when the experiment is over.

In other words, update the GUI/CLI not for every single `SnapshotUpdateEvent`, which is—for some fields—far too many to be valuable. Instead, update often, but regularly.

In other commits, PartialSnapshot testing has been changed, dnspython has now a lower bound of v2, and the `max_queue` size of the evaluator was dropped.

### CODE REVIEW

**Comment by sondreso** [comment]
Jenkins test this please

File: ert_shared/status/tracker/evaluator.py [hunk 1]
```diff
@@ -49,13 +54,16 @@ def __init__(
         self._work_queue = queue.Queue()
 
         self._drainer_thread = threading.Thread(
-            target=self._drain_monitor, name="DrainerThread"
+            target=self._drain_monitor,
+            name="DrainerThread",
+            args=(next_ensemble_evaluator_wait_time,),
```
Comment (DanSava) [inline ert_shared/status/tracker/evaluator.py:59]: Is there a special reason `next_ensemble_evaluator_wait_time` is passed as an argument to `_drain_monitor` and not made available inside the function similar to `token` or `cert`?

File: ert_shared/status/tracker/evaluator.py [hunk 2]
```diff
@@ -115,20 +124,66 @@ def _drain_monitor(self):
...
+                    if event is EvaluatorTracker.DONE:
+                        done = True
+                        break
+                    events.append(event)
+                except queue.Empty:
+                    break
+
+            if events:
+                yield from self._batch(events)
+            time.sleep(self._general_interval)
```
Comment (DanSava) [inline ert_shared/status/tracker/evaluator.py:143]: Why do we need to sleep here? and what is the expected value for the `_general_interval` ? 1 or 5?

**Comment by sondreso** [comment]
I think this looks good! But I will hold of the final review until it's not a draft anymore :smile:

**Comment by jondequinor** [comment]
This fix introduces https://github.com/equinor/ert/issues/1635. But the fix is not trivial. Either way, should it be attempted, or should we find another way to fix the lag? @sondreso @DanSava

File: ert_shared/status/tracker/evaluator.py (hunk 2 above)
Comment (jondequinor) [reply ert_shared/status/tracker/evaluator.py:143]: It varies, but yes, as we discussed without the sleep, you might end up with batches of size 1, defeating the purpose of this PR, and most like not fixing the GUI lag.

File: ert_shared/status/tracker/evaluator.py (hunk 1 above)
Comment (jondequinor) [reply ert_shared/status/tracker/evaluator.py]: Fixed!

**Comment by sondreso** [comment]
Jenkins test this please

**Comment by jondequinor** [comment]
Jenkins test this please

**Review by sondreso — APPROVED** [review]
LGTM 🚀

**Comment by jondequinor** [comment]
For jenkins failure see #1639
